java-checks/src/test/files/checks/CustomCryptographicAlgorithmCheck.java

```diff
@@ -0,0 +1,59 @@
+import java.security.MessageDigest;
+import java.util.Observable;
+
+class A { // Compliant
+}
+
+class B extends MessageDigest { // Noncompliant
```

Comment:
You can simplify your test cases by declaring B and C classes abstract (avoid the unecessary implementation of methods in the example)